Kudu.Core/Jobs/ContinuousJobRunner.cs

```diff
@@ -297,5 +302,26 @@ protected virtual void Dispose(bool disposing)
                 }
             }
         }
+
+        private int GetAvailablePortForWebJob()
+        {
+            if (JobSettings.IsWebHook)
+            {
+                var listener = new TcpListener(IPAddress.Loopback, 0);
+                try
+                {
+                    listener.Start();
+                    return ((IPEndPoint)listener.LocalEndpoint).Port;
+                }
+                finally
+                {
+                    listener.Stop();
```

Comment:
The idea is that WebJob will start listening as soon as it starts, grabbing the port. That of course leaves a small race condition window. Whether it's something that can realistically happen depends on how the ports are allocated. If each call, results in a mostly random available port, then the odds of a race are low enough that we don't care. But if the port always come sequentially, then it becomes plausible for two WebJobs (possibly in different sites) to end up with the same port.